### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by msrb**

### CODE REVIEW

**Comment by msrb** [comment]
@jimbair wanna learn some Czenglish? Here's your chance :D

File: README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+* [isProduction()](./docs/steps/isProduction.md)
+* [isPullRequest()](./docs/steps/isPullRequest.md)
+* [sendMessage()](./docs/steps/sendMessage.md)
+* [setBuildNameFromArtifactId()](./docs/steps/setBuildNameFromArtifactId.md)
+
+## Limitations
+
+Fedora CI is "artifact-centric", i.e. everything  more or less evolves around artifacts. This library currently supports only "koji-build" artifacts, modules and other artifact types are not supported yet.
+
+## Developemt
```
Comment (sturivny) [inline README.md:36]: Development

File: README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+# Fedora CI pipeline library
+
+This git repository contains a library of reusable [Jenkins Pipeline](https://jenkins.io/doc/book/pipeline/) steps and functions that can be used in your Jenkinsfile to make your Fedora CI pipelines simple.
+
+## How to use this library
+
+This library is intended to be used as a [global shared library](https://jenkins.io/doc/book/pipeline/shared-libraries/#global-shared-libraries) configured in your Jenkins instance.
+
+It is recommened to pin down a specific version of the library and update cautiously.
```
Comment (sturivny) [inline README.md:9]: It is recommended

File: README.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+```jenkinsfile
+@Library('fedora-pipeline-library') _
+```
+
+The snippet above assumes that you decided to use the name `fedora-pipeline-library` when configuring the library in the global settings.
+
+## Steps
+
+This is a list of all steps implemented in the library. Click on particular step to find our more about it.
```
Comment (sturivny) [inline README.md:21]: find out more

File: README.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+* [getGitUrl()](./docs/steps/getGitUrl.md)
+* [getReleaseIdFromBranch()](./docs/steps/getReleaseIdFromBranch.md)
+* [isProduction()](./docs/steps/isProduction.md)
+* [isPullRequest()](./docs/steps/isPullRequest.md)
+* [sendMessage()](./docs/steps/sendMessage.md)
+* [setBuildNameFromArtifactId()](./docs/steps/setBuildNameFromArtifactId.md)
+
+## Limitations
+
+Fedora CI is "artifact-centric", i.e. everything  more or less evolves around artifacts. This library currently supports only "koji-build" artifacts, modules and other artifact types are not supported yet.
```
Comment (sturivny) [inline README.md:34]: "everything more" - one whitespace should be :)

File: README.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+### Coding style
+
+Jenkinsfiles are almost Groovy and Groovy is almost Java. And Java is camelCase ;)
+
+### What belongs in here
+
+Reimplementing the world in Groovy is not what we want here. The goal is to have simple and maintainable CI pipelines.
+
+Sometimes it may be better (and more future-proof) to implement what you need in Python and just run in CI in a container. In future, when we switch to [OpenShift/Tekton Pipelines](https://www.openshift.com/learn/topics/pipelines) we will just take such Python implementation with us.
```
Comment (sturivny) [inline README.md:56]: In the future, when

File: README.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+However, if what you're trying to do involves Jenkins APIs and/or internals (i.e. it is tied to Jenkins), and it is relatively simple to do in Groovy, then putting it in here may be a good idea.
+
+Also, if your pipeline step is a function and you need use the return value from it later in the pipeline, then implementing this step as an external command in your favorite language means that you will likely need to parse standard output of that command in the pipeline. Which doesn't exactly scream simple and maintainable :)
```
Comment (sturivny) [inline README.md:60]: you need to use

File: README.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+Use your own judgement!
```
Comment (sturivny) [inline README.md:62]: judgment!

File: docs/steps/buildImageAndPushToRegistry.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
+# buildImageAndPushToRegistry() step
+
+This pipeline step builds a container image from given Git repository and pushes it to an external registry.
```
Comment (sturivny) [inline docs/steps/buildImageAndPushToRegistry.md:3]: from a given Git

File: docs/steps/getGitUrl.md [hunk 9]
```diff
@@ -0,0 +1,13 @@
+# getGitUrl() step
+
+This step returns URL of the Git repository for the current build. For non-production builds (i.e. for pull requests), it returns URL of the fork.
```
Comment (sturivny) [inline docs/steps/getGitUrl.md:3]: the URL of

File: docs/steps/isPullRequest.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
+# isPullRequest() step
+
+This step returns `true` if the current build is a build for pull request. It returns `false` otherwise.
```
Comment (sturivny) [inline docs/steps/isPullRequest.md:3]: for a pull request

File: docs/steps/sendMessage.md [hunk 11]
```diff
@@ -0,0 +1,22 @@
+# sendMessage() step
+
+This step sends a CI message to the message bus. For more information, check the [CI Messages](https://pagure.io/fedora-ci/messages) project.
+
+**Warning** This step will be probably rapidly evolving in the near future, so the API may change.
```
Comment (sturivny) [inline docs/steps/sendMessage.md:5]: Not sure about this sentence. "probably", "rapidly", "near future"

So will it be evolving or not? :)

File: README.md (hunk 2 above)
Comment (sturivny) [inline README.md:9]: update it cautiously

File: README.md (hunk 3 above)
Comment (sturivny) [inline README.md:21]: on a particular step

**Comment by jimbair** [comment]
> @jimbair wanna learn some Czenglish? Here's your chance :D

I'd be happy to help :) I'll take a look tomorrow.

File: README.md (hunk 2 above)
Comment (msrb) [reply README.md:9]: Thanks for the review @sturivny. I think I am missing some plugin in vscode - I'd expect that IDE would mark typos like these :)

Comment (sturivny) [reply README.md:9]: It's a tricky thing actually, probably you added that words like variables names =)

File: README.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+* [getGitUrl()](./docs/steps/getGitUrl.md)
+* [getReleaseIdFromBranch()](./docs/steps/getReleaseIdFromBranch.md)
+* [isProduction()](./docs/steps/isProduction.md)
+* [isPullRequest()](./docs/steps/isPullRequest.md)
+* [sendMessage()](./docs/steps/sendMessage.md)
+* [setBuildNameFromArtifactId()](./docs/steps/setBuildNameFromArtifactId.md)
+
+## Limitations
+
+Fedora CI is "artifact-centric", i.e. everything more or less evolves around artifacts. This library currently supports only "koji-build" artifacts, modules and other artifact types are not supported yet.
```
Comment (jimbair) [inline README.md:34]: I think this requires a semi-colon:

"This library currently supports only "koji-build" artifacts; modules and other artifact types are not supported yet."

**Comment by jimbair** [comment]
Aside from my one minor nit above, it looks great!

File: README.md (hunk 12 above)
Comment (msrb) [reply README.md]: Updated. Thanks! :wink:

**Comment by msrb** [comment]
Thanks for the review @sturivny and @jimbair :)

**Pull request merged**